ofrak_core/ofrak/gui/server.py

```diff
@@ -739,6 +742,57 @@ async def search_for_vaddr(self, request: Request) -> Response:
         except NotFoundError:
             return json_response([])
 
+    @exceptions_to_http(SerializedError)
+    async def search_for_string(self, request: Request):
+        resource = await self._get_resource_for_request(request)
+        body = await request.json()
+        string_query_string = body["search_query"]
+        if string_query_string == "":
+            return json_response(None)
+        regex = body["regex"]
+        case_ignore = body["case_ignore"]
+        if not isinstance(string_query_string, str):
+            raise ValueError("Invalid search query.")
+        string_query: Union[bytes, re.Pattern[bytes]] = string_query_string.encode()
+        if case_ignore:
+            string_query = re.compile(string_query, re.IGNORECASE)
+        elif regex:
+            string_query = re.compile(string_query)
+
+        offsets = await resource.search_data(string_query)
+        found_resources = []
+        if len(offsets) > 0:
+            found_resources.append(resource.get_id().hex())
+        for child in await resource.get_descendants():
+            offsets = await child.search_data(string_query)
```

Comment:
```suggestion
            if child.get_data_id() is None:
                continue
            offsets = await child.search_data(string_query)
```
Handles resources that don't have data, otherwise an AttributeError gets raised.